### PR TITLE
RAI-15628 Add beta version of snowflake datastream apis

### PR DESCRIPTION
**Pull request by priti-pa**

### CODE REVIEW

File: rai/models.go [hunk 1]
```diff
@@ -456,6 +456,20 @@ type SnowflakeDataStream struct {
 }
 
 type createSnowflakeDataStreamRequest struct {
+	Snowflake struct {
+		Database  string `json:"database"` // native app db
```
Comment (wienleung) [inline rai/models.go:460]: same as other PR, we don't need Database and Schema (previously for DB Link) and Role and Warehouse (previously for executing data stream SQL on the integration services?)

Comment (priti-pa) [reply rai/models.go]: I have removed db, schema, role and warehouse.

**Review by wienleung — APPROVED** [review]

File: rai/client.go [hunk 2]
```diff
@@ -1836,7 +1838,75 @@ func (c *Client) GetSnowflakeDataStreamStatus(
...
+// beta
+
+type DataStreamOpts struct {
+	RaiDatabase string
+	Relation    string
+	ObjectName  string
+}
+
+// Register snowflake datastream created by native app to replicate data from a Snowflake table/view to a RAI relation.
+func (c *Client) RegisterSnowflakeDataStream(
```
Comment (rsulli-rai) [inline rai/client.go:1857]: are these intended for a customer or RAI use? does the caller need a rai token from the linked account?

Comment (priti-pa) [reply rai/client.go:1857]: These are intended for RAI's use; the proxy will call into the integration service endpoint using these.

Comment (rsulli-rai) [reply rai/client.go:1857]: whats is preventing client users form calling these endpoints? seems like a potential risk

Comment (priti-pa) [reply rai/client.go:1857]: They can't call these endpoints without an RAI token. To generate the token, they need to know the account associated with the integration and have access to that account. Please correct me if I am missing a point here.

**Pull request merged**